File: Analysis of Algorithms/Project2 - analo of alg/haversine.c
/*#include <stdio.h>
#include <math.h>

double haversine(double lat1, double long1, double lat2, double long2)
{
	double radiusOfEarthInMiles = 3958.8;
	
	double degreesToRadians = M_PI / 180.0;
	
	//Convert degrees to radians:
	lat1 *= degreesToRadians;
	long1 *= degreesToRadians;
	lat2 *= degreesToRadians;
	long2 *= degreesToRadians;
	
	double a = pow(sin((lat2-lat1)/2), 2) + cos(lat1)*cos(lat2)*pow(sin((long2-long1)/2), 2);
	double c = 2*atan2(sqrt(a), sqrt(1-a));
	
	return radiusOfEarthInMiles*c;
}

int main(){
	
	double lat1 = 29.5827351;
	double long1 = -98.621094;

	double lat2 = 29.4259671;
	double long2 = -98.4883306;

	double dist = haversine(lat1,long1,lat2,long2);

	printf("The distance between UTSA Main campus: (%f,%f) and the Alamo: (%f,%f) is %f miles.\n",lat1,long1,lat2,long2,dist);
	
}
*/

File: Analysis of Algorithms/Project2 - analo of alg/haversine.py
#!/usr/bin/env python3
import math

def haversine(lat1, long1, lat2, long2):
	
	radiusOfEarthInMiles = 3958.8
	
	lat1Radians = math.radians(lat1)
	long1Radians = math.radians(long1)
	lat2Radians = math.radians(lat2)
	long2Radians = math.radians(long2)
	
	a = pow(math.sin((lat2Radians-lat1Radians)/2), 2) + math.cos(lat1Radians)*math.cos(lat2Radians)*pow(math.sin((long2Radians-long1Radians)/2), 2)
	c = 2*math.atan2(math.sqrt(a), math.sqrt(1-a))
	
	return radiusOfEarthInMiles*c
	
	
lat1 = 29.5827351
long1 = -98.621094

lat2 = 29.4259671
long2 = -98.4883306

dist = haversine(lat1,long1,lat2,long2)

print("The distance between UTSA Main campus: ("+str(lat1)+","+str(long1)+") and the Alamo: ("+str(lat2)+","+str(long2)+") is " + str(dist) + " miles.")